api/src/controllers/reviews.ts

```diff
@@ -191,77 +168,62 @@ const reviewsRouter = router({
+      if (input.vote === 0) {
+        await db.delete(vote).where(and(eq(vote.userId, ctx.session.userId!), eq(vote.reviewId, input.id)));
+        return true;
+      }
 
-      return { deltaScore: deltaScore };
-    } else {
-      //no old vote, just add in new vote data
-      await Review.updateOne({ _id: id }, { $inc: { score: deltaScore } });
-      //sends in vote
-      await new Vote({ userID: ctx.session.passport!.user.id, reviewID: id, score: deltaScore }).save();
-      return { deltaScore: deltaScore };
-    }
-  }),
+      await db
+        .insert(vote)
+        .values({ userId: ctx.session.userId!, reviewId: input.id, vote: input.vote })
+        .onConflictDoUpdate({ target: [vote.userId, vote.reviewId], set: { vote: input.vote } });
+    }),
 
-  verify: adminProcedure.input(z.object({ id: z.string() })).mutation(async ({ input }) => {
-    return await Review.updateOne({ _id: input.id }, { verified: true });
+  verify: adminProcedure.input(z.object({ id: z.number() })).mutation(async ({ input }) => {
+    await db.update(review).set({ verified: true }).where(eq(review.id, input.id));
+    return true;
   }),
 
   edit: userProcedure.input(editReviewSubmission).mutation(async ({ input, ctx }) => {
-    if (!(await userWroteReview(ctx.session.passport!.user.id, input._id))) {
+    if (!(await userWroteReview(ctx.session.userId, input.id))) {
       throw new TRPCError({ code: 'UNAUTHORIZED', message: 'You are not the author of this review.' });
     }
 
-    const { _id, ...updateWithoutId } = input;
-    await Review.updateOne({ _id }, updateWithoutId);
-    return input;
+    const { id, ...updateWithoutId } = input;
+    await db.update(review).set(updateWithoutId).where(eq(review.id, id));
+    return true;
   }),
 
   /**
    * Get featured review for a course or professor
    */
   featured: publicProcedure.input(featuredQuery).query(async ({ input }) => {
-    // search by professor or course field
-    let field = '';
-    if (input.type == 'course') {
-      field = 'courseID';
-    } else if (input.type == 'professor') {
-      field = 'professorID';
-    }
-
-    // find first review with the highest score
-    const review = await Review.findOne<ReviewData>({ [field]: input.id })
-      .sort({ reviewContent: -1, score: -1, verified: -1 })
-      .limit(1);
-
-    return review;
+    const voteSubQuery = db
+      .select({ reviewId: vote.reviewId, score: sql`sum(${vote.vote})`.mapWith(Number).as('score') })
+      .from(vote)
+      .groupBy(vote.reviewId)
+      .as('vote_query');
+
+    const field = input.type === 'course' ? review.courseId : review?.professorId;
+    const featuredReview = (
+      await db
+        .select()
+        .from(review)
+        .where(eq(field, input.id))
+        .leftJoin(voteSubQuery, eq(voteSubQuery.reviewId, review.id))
+        .orderBy(desc(review.content), desc(voteSubQuery.score), desc(review.verified)) // featured review criteria
```

Comment:
Put as a variable instead of as a comment

`const featuredReviewCriteria = [desc(review.content), desc(voteSubQuery.score), desc(review.verified)`

`.orderBy(...featuredReviewCriteria)`